aioshelly/common.py

```diff
@@ -137,3 +139,29 @@ def shelly_supported_firmware(result: dict[str, Any]) -> bool:
     # We compare firmware release dates because Shelly version numbering is
     # inconsistent, sometimes the word is used as the version number.
     return int(match[0]) >= fw_ver
+
+
+async def trigger_ota_http(
+    session: aiohttp.ClientSession,
+    host: str,
+    gen: int,
+    auth: aiohttp.BasicAuth | None = None,
+) -> bool:
+    """Trigger a firmware update via OTA http endpoint."""
+    if gen in BLOCK_GENERATIONS:
+        path = "ota"
+    else:
+        path = "rpc/Shelly.Update"
+    try:
+        await session.get(
+            URL.build(scheme="http", host=host, path=f"/{path}"),
+            auth=auth,
```

Comment:
This will fail if used for Gen2

Reply:
To avoid complexity of the code, we should move the Auth management out of the class so to reuse it here and there

Reply:
The auth code is part of a class so you would have to do many changes for this and you end up with a code that was supposed to serve for wsrpc under common while it is nothing in common with Gen1.

I really don't understand were you are going with it. We enforce a minimum version which blocks us from initializing a device (although any minimum you set for Gen2 will not block you from initializing the device since there are no breaking API changes) than we move code to common so we can talk to the device without initializing it while we shouldn't have gone there in the first place.